src/main/scala/java/time/Instant.scala

```diff
@@ -0,0 +1,267 @@
+
+  def plusNanos(nans: Long): Instant = plus(0, nans)
+
+  private def plus(secs: Long, nans: Long): Instant = {
+    if (secs == 0 && nans == 0) this
+    else {
+      val secondsFromNanos = Math.floorDiv(nans, NANOS_IN_SECOND)
+      val remainingNanos = Math.floorMod(nans, NANOS_IN_SECOND)
+      val additionalSecs = Math.addExact(secs, secondsFromNanos)
+      ofEpochSecond(
+        Math.addExact(seconds, additionalSecs),
+        Math.addExact(nanos, remainingNanos)
+      )
+    }
+  }
+
+  override def minus(amount: Long, unit: TemporalUnit): Instant = {
+    if (amount == Long.MinValue) plus(Long.MaxValue, unit).plus(1, unit)
+    else plus(-amount, unit)
+  }
+
+  def minusSeconds(secs: Long): Instant = minus(secs, SECONDS)
+
+  def minusMillis(millis: Long): Instant = minus(millis, MILLIS)
+
+  def minusNanos(nans: Long): Instant = minus(nans, NANOS)
+
+  // Not implemented
+  // def query[R](query: TemporalQuery[R]): R
+
+  def adjustInto(temporal: Temporal): Temporal =
+    temporal.`with`(INSTANT_SECONDS, seconds).`with`(NANO_OF_SECOND, nanos)
+
+  def until(end: Temporal, unit: TemporalUnit): Long = {
+    val endInstant = from(end)
+    val nanosDiff = endInstant.totalNanos - totalNanos
+    unit match {
+      case NANOS     => nanosDiff
+      case MICROS    => nanosDiff / NANOS_IN_MICRO
+      case MILLIS    => nanosDiff / NANOS_IN_MILLI
+      case SECONDS   => nanosDiff / NANOS_IN_SECOND
+      case MINUTES   => nanosDiff / NANOS_IN_MINUTE
+      case HOURS     => nanosDiff / NANOS_IN_HOUR
+      case HALF_DAYS => nanosDiff / (NANOS_IN_HOUR * 12)
+      case DAYS      => nanosDiff / NANOS_IN_DAY
+
+      case _: ChronoUnit =>
+        throw new UnsupportedTemporalTypeException(s"Unit not supported: $unit")
+
+      case _ => unit.between(this, end)
+    }
+  }
+
+  // Not implemented
+  // def atOffset(offset: ZoneOffset): OffsetDateTime
+
+  // Not implemented
+  // def atZone(zone: ZoneId): ZonedDateTime
+
+  def toEpochMilli(): Long = Math.floorDiv(totalNanos, NANOS_IN_MILLI)
+
+  def compareTo(that: Instant): Int =
+    totalNanos.compareTo(that.totalNanos)
+
+  def isAfter(that: Instant): Boolean = compareTo(that) > 0
+
+  def isBefore(that: Instant): Boolean = compareTo(that) < 0
+
+  override def equals(other: Any): Boolean = other match {
+    case that: Instant => totalNanos == that.totalNanos
+    case _             => false
+  }
+
+  override def hashCode(): Int = totalNanos.hashCode()
+
+  override def toString: String = {
+    val date = new js.Date(toEpochMilli().toDouble)
+    date.toISOString()
+  }
+
+  // Not implemented
+  // def format(format: DateTimeFormatter): String
+
+}
+
+object Instant {
+  import Constants._
+  import ChronoField._
+
+  final val EPOCH = new Instant(0, 0)
+
+  private val MinSecond = -31557014167219200L
+  private val MaxSecond = 31557014167219200L
+
+  final val MIN = ofEpochSecond(MinSecond)
+  final val MAX = ofEpochSecond(MaxSecond)
```

Comment:
Here is where you are getting the exception on the JVM.

```scala> 
Instant.ofEpochSecond(31557014167219200L)
java.time.DateTimeException: Instant exceeds minimum or maximum instant
  at java.time.Instant.create(Instant.java:411)
  at java.time.Instant.ofEpochSecond(Instant.java:304)
  ... 33 elided
```

The spec says that the value is `The maximum supported Instant, '1000000000-12-31T23:59:59.999999999Z'.`, therefore it will need a definition involving nanos.